planemo/galaxy/config.py

```diff
@@ -1003,84 +978,6 @@ def _database_connection(database_location, **kwds):
     return database_connection
 
 
-def attempt_database_preseed(
-    ctx, effective_galaxy_root, database_location, **kwds
-):
-    """If database location is unset, attempt to seed the database."""
-    if os.path.exists(database_location):
-        # Can't seed an existing database.
-        return False
-
-    if not _database_connection(database_location, **kwds).startswith("sqlite"):
-        # Not going to use an sqlite database, don't preseed.
-        return False
-
-    preseeded_database = True
-    galaxy_sqlite_database = kwds.get("galaxy_database_seed", None)
-    galaxy_branch = kwds.get("galaxy_branch", None)
-    try:
-        _download_database_template(
-            ctx,
-            effective_galaxy_root,
-            database_location,
-            galaxy_sqlite_database=galaxy_sqlite_database,
-            galaxy_branch=galaxy_branch
-        )
-    except Exception as e:
-        print(e)
-        # No network access - just roll forward from null.
-        preseeded_database = False
-    return preseeded_database
-
-
-def _download_database_template(
-    ctx,
-    galaxy_root,
-    database_location,
-    galaxy_sqlite_database=None,
-    galaxy_branch=None,
-):
-
-    if galaxy_sqlite_database is not None:
-        shutil.copyfile(galaxy_sqlite_database, database_location)
-        return True
-
-    download_migration = None
-    newest_migration = _newest_migration_version(galaxy_root, galaxy_branch)
-    for migration in DOWNLOADABLE_MIGRATION_VERSIONS:
-        if newest_migration >= migration:
-            download_migration = migration
-            break
-
-    if download_migration:
-        download_name = "db_gx_rev_0%d.sqlite" % download_migration
-        download_url = DOWNLOADS_URL + download_name
-        ctx.cache_download(download_url, database_location)
```

Comment:
The `Context.cache_download()` method (defined in `planemo/cli.py`) is now unused, not sure if we want to drop that too.

Reply:
I may reuse that for downloading the prebuilt galaxy client (https://github.com/mvdbeek/galaxy-client-cache/blob/master/.github/workflows/build_client.yaml)